File: docs/vsto/codesnippet/VisualBasic/Trin_VstcoreProgrammingExcelVB/Sheet1.vb
Public Class Sheet1

    Private Sub Sheet1_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup
    End Sub

    Private Sub Sheet1_Shutdown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shutdown
    End Sub


    '<Snippet1>
    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles Button1.Click
        Globals.Sheet1.NamedRange1.Value2 = Me.TextBox1.Text
    End Sub
    '</Snippet1>


    Sub Test1()
        '<Snippet2>
        Dim NamedRange1 As Microsoft.Office.Tools.Excel.NamedRange = _
            Me.Controls.AddNamedRange(Me.Range("A1"), "NamedRange1")
        '</Snippet2>

        '<Snippet3>
        Dim width As Double = CType(NamedRange1.ColumnWidth, Double)
        '</Snippet3>

        '<Snippet4>
        MessageBox.Show("Column width: " & width.ToString())
        '</Snippet4>
    End Sub

    Sub TestDynamic()
        '<Snippet9>
        Dim castRange As Excel.Range = CType(Globals.Sheet1.Cells(1, 1), Excel.Range)
        '</Snippet9>

        '<Snippet10>
        Dim dynamicRange As Excel.Range = Globals.Sheet1.Cells(1, 1)
        '</Snippet10>
    End Sub
End Class

File: docs/vsto/codesnippet/CSharp/Trin_VstcoreProgrammingExcelCS/Sheet1.cs
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Excel=Microsoft.Office.Interop.Excel;
using Office=Microsoft.Office.Core;

namespace Trin_VstcoreProgrammingExcelCS
{
    public partial class Sheet1
    {
        private void Sheet1_Startup(object sender, System.EventArgs e)
        {
            Test1();
        }

        private void Sheet1_Shutdown(object sender, System.EventArgs e)
        {
        }


        //<Snippet1>
        private void button1_Click(object sender, EventArgs e)
        {
            Globals.Sheet1.namedRange1.Value2 = this.textBox1.Text;
        }
        //</Snippet1>


        private void Test1()
        {
            //<Snippet2>
            Microsoft.Office.Tools.Excel.NamedRange NamedRange1 = 
                this.Controls.AddNamedRange(this.Range["A1"], "NamedRange1");
            //</Snippet2>
        
            //<Snippet3>
            double width = (double)NamedRange1.ColumnWidth;
            //</Snippet3>

            //<Snippet4>
            MessageBox.Show("Column width: " + width.ToString());
            //</Snippet4>
        }

        private void TestDynamic()
        {
           
            Excel.Range castRange = (Excel.Range)Globals.Sheet1.Cells[1, 1];
       

            //<Snippet10>
            Excel.Range dynamicRange = Globals.Sheet1.Cells[1, 1];
            //</Snippet10>
        }


        private void InternalStartup()
        {
            this.Startup+=new System.EventHandler(Sheet1_Startup);
            this.Shutdown+=new System.EventHandler(Sheet1_Shutdown);

            this.button1.Click += new EventHandler(button1_Click);

        }
    }
}
